toolkits/search/arcade_search/utils.py

```diff
@@ -51,6 +63,123 @@ def call_serpapi(context: ToolContext, params: dict) -> dict:
+    distance_unit: GoogleMapsDistanceUnit = DEFAULT_GOOGLE_MAPS_DISTANCE_UNIT,
+    travel_mode: GoogleMapsTravelMode = DEFAULT_GOOGLE_MAPS_TRAVEL_MODE,
+) -> dict:
+    """Get directions from Google Maps.
+
+    Provide either all(origin_address, destination_address) or
+    all(origin_latitude, origin_longitude, destination_latitude, destination_longitude).
+
+    Args:
+        context: Tool context containing required Serp API Key secret.
+        origin_address: Origin address.
+        destination_address: Destination address.
+        origin_latitude: Origin latitude.
+        origin_longitude: Origin longitude.
+        destination_latitude: Destination latitude.
+        destination_longitude: Destination longitude.
+        language: Language to use in the Google Maps search. Defaults to 'en' (English).
+        country: 2-letter country code to use in the Google Maps search. Defaults to None
+            (no country is specified).
+        distance_unit: Distance unit to use in the Google Maps search. Defaults to 'km'
+            (kilometers).
+        travel_mode: Travel mode to use in the Google Maps search. Defaults to 'best'
+            (best mode).
+
+    Returns:
+        The directions from Google Maps.
+    """
+    language = language.lower()
+
+    if language not in LANGUAGE_CODES:
+        raise LanguageNotFoundError(language)
+
+    params = prepare_params(
+        engine="google_maps_directions",
+        hl=language,
+        distance_unit=google_maps_distance_unit_to_serpapi(distance_unit),
+        travel_mode=google_maps_travel_mode_to_serpapi(travel_mode),
+    )
+
+    if any([
+        origin_latitude,
+        origin_longitude,
+        destination_latitude,
+        destination_longitude,
+    ]) and any([origin_address, destination_address]):
+        raise ValueError("Either coordinates or addresses must be provided, not both")
+
+    elif all([origin_latitude, origin_longitude, destination_latitude, destination_longitude]):
+        params["start_coords"] = f"{origin_latitude},{origin_longitude}"
+        params["end_coords"] = f"{destination_latitude},{destination_longitude}"
+
+    elif all([origin_address, destination_address]):
+        params["start_addr"] = str(origin_address)
+        params["end_addr"] = str(destination_address)
+
+    else:
+        raise ValueError("Either coordinates or addresses must be provided")
+
+    if country:
+        country = country.lower()
+        if country not in COUNTRY_CODES:
+            raise CountryNotFoundError(country)
+        params["gl"] = country
+
+    results = call_serpapi(context, params)
+
+    for direction in results["directions"]:
+        if "arrive_around" in direction:
+            direction["arrive_around"] = enrich_google_maps_arrive_around(
+                direction["arrive_around"]
+            )
+
+    return results
+
+
+def google_maps_travel_mode_to_serpapi(travel_mode: GoogleMapsTravelMode) -> int:
```

Comment:
See [this enum](https://github.com/ArcadeAI/arcade-ai/blob/25c9f76ca8de49a0d1205da5337a7f9f6ace11fb/toolkits/search/arcade_search/enums.py#L67) for an alternative approach to converting to SerpAPI value.

Your approach works just fine, but wanted you to be aware of alternatives.